internal/stores/objects_test.go

```diff
@@ -559,53 +528,240 @@ func TestContractSectors(t *testing.T) {
+		for _, h := range hosts {
+			ids = append(ids, h.PublicKey)
+		}
+		return
+	}
+
+	// fetch inserted slab
+	inserted := fetchSlab()
+
+	// assert both sectors were upload to one contract/host
+	for i := 0; i < 2; i++ {
+		if cids := contractIds(inserted.Shards[i].DBSector.Contracts); len(cids) != 1 {
+			t.Fatalf("sector %d was uploaded to unexpected amount of contracts, %v!=1", i+1, len(cids))
+		} else if hks := hostKeys(inserted.Shards[i].DBSector.Hosts); len(hks) != 1 {
+			t.Fatalf("sector %d was uploaded to unexpected amount of hosts, %v!=1", i+1, len(hks))
+		}
+	}
+
+	// create a contract set using contracts for h1 and h3 (h2 is bad)
+	err = db.SetContractSet("autopilot", []types.FileContractID{fcid1, fcid3})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// fetch slabs for migration and assert there is only one
+	toMigrate, err := db.SlabsForMigration("autopilot", -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(toMigrate) != 1 {
+		t.Fatal("unexpected number of slabs to migrate", len(toMigrate))
+	}
+
+	// migrate the sector from h2 to h3
+	slab := toMigrate[0]
+	slab.Shards[1] = object.Sector{
+		Host: hk3,
+		Root: consensus.Hash256{2},
+	}
+
+	// update the slab to reflect the migration
+	err = db.PutSlab(slab, map[consensus.PublicKey]types.FileContractID{
+		hk1: fcid1,
+		hk3: fcid3,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// fetch updated slab
+	updated := fetchSlab()
+
+	// assert the first sector is still only on one host, also assert it's h1
+	if cids := contractIds(updated.Shards[0].DBSector.Contracts); len(cids) != 1 {
+		t.Fatalf("sector 1 was uploaded to unexpected amount of contracts, %v!=1", len(cids))
+	} else if cids[0] != fcid1 {
+		t.Fatal("sector 1 was uploaded to unexpected contract", cids[0])
+	} else if hks := hostKeys(updated.Shards[0].DBSector.Hosts); len(hks) != 1 {
+		t.Fatalf("sector 1 was uploaded to unexpected amount of hosts, %v!=1", len(hks))
+	} else if hks[0] != hk1 {
+		t.Fatal("sector 1 was uploaded to unexpected host", hks[0])
+	}
+
+	// assert the second sector however is uploaded to two hosts, assert it's h2 and h3
+	if cids := contractIds(updated.Shards[1].DBSector.Contracts); len(cids) != 2 {
+		t.Fatalf("sector 1 was uploaded to unexpected amount of contracts, %v!=2", len(cids))
+	} else if cids[0] != fcid2 || cids[1] != fcid3 {
+		t.Fatal("sector 1 was uploaded to unexpected contracts", cids[0], cids[1])
+	} else if hks := hostKeys(updated.Shards[1].DBSector.Hosts); len(hks) != 2 {
+		t.Fatalf("sector 1 was uploaded to unexpected amount of hosts, %v!=2", len(hks))
+	} else if hks[0] != hk2 || hks[1] != hk3 {
+		t.Fatal("sector 1 was uploaded to unexpected hosts", hks[0], hks[1])
+	}
+
+	// fetch slabs for migration and assert there are none left
+	toMigrate, err = db.SlabsForMigration("autopilot", -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(toMigrate) != 0 {
+		t.Fatal("unexpected number of slabs to migrate", len(toMigrate))
+	}
```

Comment:
Might have missed it but can you also check a few more things?

- dbSlab still only got 1 entry
- fetching the object should still return the right object with the updated slab